servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletHttpRequest.java

```diff
@@ -0,0 +1,603 @@
+            }
+        }
+        if (error) {
+            logger.info("Can't parse " + getHeader(name) + " to date");
+        }
+        return -1;
+    }
+
+    @Override
+    @Nullable
+    public String getHeader(String name) {
+        requireNonNull(name, "name");
+        return httpRequest.headers().get(name);
+    }
+
+    @Override
+    public Enumeration<String> getHeaderNames() {
+        return Collections.enumeration(
+                httpRequest.headers().names().stream()
+                           .map(AsciiString::toString).collect(ImmutableList.toImmutableList()));
+    }
+
+    @Override
+    public StringBuffer getRequestURL() {
+        return new StringBuffer(httpRequest.scheme() + "://" + httpRequest.authority() + requestURI);
+    }
+
+    @Override
+    @Nullable
+    public String getPathInfo() {
+        return pathInfo;
+    }
+
+    @Override
+    @Nullable
+    public String getQueryString() {
+        return serviceRequestContext.query();
+    }
+
+    @Override
+    public String getRequestURI() {
+        return requestURI;
+    }
+
+    @Override
+    public String getServletPath() {
+        return servletPath;
+    }
+
+    @Override
+    public HttpSession getSession(boolean create) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public HttpSession getSession() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public String changeSessionId() {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public Enumeration<String> getHeaders(String name) {
+        requireNonNull(name, "name");
+        return Collections.enumeration(
+                httpRequest.headers().getAll(name).stream().collect(ImmutableList.toImmutableList()));
+    }
+
+    @Override
+    public int getIntHeader(String name) {
+        requireNonNull(name, "name");
+        final String headerStringValue = getHeader(name);
+        if (isNullOrEmpty(headerStringValue)) {
+            return -1;
+        }
+        return Integer.parseInt(headerStringValue);
+    }
+
+    @Override
+    public String getMethod() {
+        return httpRequest.method().toString();
+    }
+
+    @Override
+    public String getContextPath() {
+        return servletContext.getContextPath();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdValid() {
+        return sessionIdSource == SessionTrackingMode.COOKIE ||
+               sessionIdSource == SessionTrackingMode.URL;
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromCookie() {
+        return sessionIdSource == SessionTrackingMode.COOKIE;
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromURL() {
+        return isRequestedSessionIdFromUrl();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromUrl() {
+        return sessionIdSource == SessionTrackingMode.URL;
```

Comment:
Shouldn't we return `false` at the moment because this method checks whether the requested session ID was conveyed to the server as part of the request URL?